knative-operator/pkg/controller/knativeservingobsolete/knativeservingobsolete_controller.go

```diff
@@ -170,8 +173,24 @@ func (r *ReconcileKnativeServingObsolete) reconcileNewResource(old *obsolete.Kna
 	return new, err
 }
 
+// Delete obsolete monitoring resources from previous version to avoid fatal ownerRef issues.
+func (r *ReconcileKnativeServingObsolete) deleteObsoleteMonitoringResources() error {
+	const path = "deploy/resources/obsolete/monitoring-1.3.0.yaml"
+
+	manifest, err := mf.NewManifest(path, false, r.client)
+	if err != nil {
+		return err
+	}
+
+	if err := manifest.DeleteAll(); err != nil {
+		log.Info("Ignoring failed deletion of obsolete monitoring resources")
```

Comment:
Are we sure we want to ignore delete failures? If we do and this bad ownerRef doesn't get deleted, we have a problem.

Reply:
Fair point, I did this to ignore NotFound errors and thelike. Manifestival seems to do the right thing for that already though, so gonna change this. Thanks!

Reply:
confirmed, MF will eat the NotFounds but return any errors that might prevent deletion